lib/requests/create-account-request.js

```diff
@@ -115,30 +116,28 @@ class CreateAccountRequest extends AuthRequest {
   /**
    * Creates an account for a given user (from a POST to `/api/accounts/new`)
    *
-   * @throws {Error} An http 400 error if an account already exists
+   * @throws {Error} If errors were encountering while validating the username.
    *
    * @return {Promise<UserAccount>} Resolves with newly created account instance
    */
-  createAccount () {
+  async createAccount () {
     let userAccount = this.userAccount
     let accountManager = this.accountManager
 
-    return Promise.resolve(userAccount)
-      .then(this.cancelIfUsernameInvalid.bind(this))
-      .then(this.cancelIfAccountExists.bind(this))
-      .then(this.createAccountStorage.bind(this))
-      .then(this.saveCredentialsFor.bind(this))
-      .then(this.sendResponse.bind(this))
-      .then(userAccount => {
-        // 'return' not used deliberately, no need to block and wait for email
-        if (userAccount && userAccount.email) {
-          debug('Sending Welcome email')
-          accountManager.sendWelcomeEmail(userAccount)
-        }
-      })
-      .then(() => {
-        return userAccount
-      })
+    this.cancelIfUsernameInvalid(userAccount)
+    this.cancelIfBlacklistedUsername(userAccount)
+    await this.cancelIfAccountExists(userAccount)
```

Comment:
Is that an [information leak](http://projects.webappsec.org/w/page/13246936/Information%20Leakage)?
That is, do you have a [protection against brute force the usernames](https://www.owasp.org/index.php/Blocking_Brute_Force_Attacks) in place?

Reply:
You're thinking that people could brute force to find out which usernames are taken? But that is already publicly available as it is now... i.e. WebIDs are not publicly listed in an index, but you could brute force through a list of username given a specific WebID-pattern to find out which usernames have WebIDs on a given server as all PODs have public "frontpage" as of now - that might change in the future.

I'm not familiar with how we handle brute force attempts at logging in with a given username though, if we handle it at all right now... (maybe @kjetilk, @RubenVerborgh, or @dmitrizagidulin know?) Definitely something we need to have in place before version 6.0.0 at least, and we should probably create a separate issue for it (I'll wait for response by the people tagged first).

Reply:
We do not currently handle brute force attacks to find out what usernames are taken. Like you said, it's publicly available information (you can check to see if an account is taken via an API call).

As far as login brute force attacks, `bcrypt` password library provides a decent protection against those (it may make sense to increase our bcrypt params to take slightly longer).